### PR TITLE
#1634 - grunt build task: minification for libraries folder and json files

### PR DESCRIPTION
**Pull request by barry-learningpool**
https://github.com/adaptlearning/adapt_framework/pull/1634 Added UglifyJS library to package.json. Added 'libraries' configuration for 'uglify' task and calling it as part of the standard 'build' task. Files which are already minified are ignored. Files, filenames and extensions are all preserved (as these this would break the requires in scriptLoader.js)

### CODE REVIEW

**Comment by oliverfoster** [comment]
which libraries have we got that are minified but don't have the .min.js ending? cos that'll reduce the time quite a bit.

**Comment by barry-learningpool** [comment]
@oliverfoster I'm in the process of renaming the files, I've had to update the modernizr.js reference in index.html, however this references also exists in the bundled Spoor HTML files. I can go ahead and create a Spoor PR for these changes, I'm just wondering about versioning. Does this mean there needs to be a framework version bump and the framework version then updated in Spoor?

**Comment by oliverfoster** [comment]
Yup, spoor will need it's minimum framework version setting also.

**Comment by moloko** [comment]
@barry-learningpool sorry but could you leave modernizr alone for now please? we can sort that out when we switch to v3. just a bit concerned about the risk of changing that i.e. someone could update framework but not spoor and then they'd have a broken course.

**Comment by moloko** [comment]
@barry-learningpool thanks for the amends, there's a conflict on scriptLoader.js though I'm afraid...

**Comment by moloko** [comment]
also you have renamed modernizr to modernizr.min.js - but we need that to be left as modernizr.js to avoid having to update the references in the various .html files

**Comment by barry-learningpool** [comment]
@moloko issues have been resolved

**Comment by oliverfoster** [comment]
core/js/libraries/modernizr.min.js needs renaming

File: grunt/config/json-minify.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = function (grunt, options) {
+    return {
+        build: {
```
Comment (oliverfoster) [inline grunt/config/json-minify.js]: can you rename from ``build`` to ``minify`` so the ``minify`` task is responsible for both json and javascript?

File: grunt/config/uglify.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = function (grunt, options) {
+    return {
+        libraries: {
+            files: [{
+                expand: true,
+                cwd: '<%= outputdir %>libraries/',
+                src: ['*.js', '!*.min.js'],
+                dest: '<%= outputdir %>libraries/'
+            }]
```
Comment (oliverfoster) [inline grunt/config/uglify.js]: Could you also add these, for the ``templates.js`` and the ``scriptLoader.js``
```javascript
            ,{
                expand: true,
                cwd: '<%= outputdir %>/adapt/js',
                src: ['*.js', '!*.min.js'],
                dest: '<%= outputdir %>/adapt/js'
            },{
                expand: true,
                cwd: '<%= outputdir %>',
                src: ['*.js', '!*.min.js'],
                dest: '<%= outputdir %>'
            }
```

**Comment by oliverfoster** [comment]
ta @barry-learningpool

File: grunt/config/uglify.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+module.exports = function (grunt, options) {
+    return {
+        libraries: {
```
Comment (moloko) [inline grunt/config/uglify.js]: you need to add 
```js
options: {
    ie8: true
},
```
here or the minified version of the code won't load in IE8. We can take this out again once we move to v3 and drop IE8 support.

**Pull request merged**